llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -429,6 +430,15 @@ void InferAddressSpacesImpl::collectRewritableIntrinsicOperands(
     appendsFlatAddressExpressionToPostorderStack(II->getArgOperand(0),
                                                  PostorderStack, Visited);
     break;
+  case Intrinsic::is_constant: {
+    Value *Ptr = II->getArgOperand(0);
+    if (Ptr->getType()->isPtrOrPtrVectorTy()) {
+      appendsFlatAddressExpressionToPostorderStack(Ptr, PostorderStack,
+                                                   Visited);
+    }
```

Comment:
unnecessary braces?

Reply:
Multiple lines

Reply:
multiple lines is not listed as an excuse in the coding style

Reply:
"However, braces should be used in cases where the omission of braces harm the readability and maintainability of the code."

and in my judgment this covers any situation with multiple lines

Reply:
How could it hurt readability?

Reply:
Because unlike a compiler I read by lines, and not statements. 

The length of that section of the style guide is absurd. It should never be wrong to include braces. That's a lot simpler than parsing out whatever that is

Reply:
> It should never be wrong to include braces.

Having to deal with google style that demands braces everywhere, and LLVM which does not want them, my personal choice is "whatever the style guide says". It may not always be a perfect choice, but  it's not worth anyone's time to argue over specific instances, where the right choice is ambiguous or is a matter of personal preference. I wish we could delegate braces/no-braces decisions to clang-format, too, but I don't think it currently handles that.

I'd stick with the style guide defaults and either have the braces removed, or a comment added to the body. Perhaps, making the function name shorter, and avoiding line-wrapping would address your readability concerns about braces/no-braces here, too.